### PR TITLE
Changes to match the introduction of the AbstractMessageChannelBinder

### PR DESCRIPTION
**Pull request by mbogoevici**
Requires https://github.com/spring-cloud/spring-cloud-stream/pull/589 to be merged first

### CODE REVIEW

File: spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java [hunk 1]
```diff
@@ -192,30 +175,90 @@ public void onInit() {
...
+		int maxAttempts = properties.getMaxAttempts();
+		if (maxAttempts > 1 || properties.getExtension().isRepublishToDlq()) {
+			RetryOperationsInterceptor retryInterceptor = RetryInterceptorBuilder.stateless()
+					.maxAttempts(maxAttempts)
+					.backOffOptions(properties.getBackOffInitialInterval(),
+							properties.getBackOffMultiplier(),
+							properties.getBackOffMaxInterval())
+					.recoverer(determineRecoverer(baseQueueName, properties.getExtension().getPrefix(),
+							properties.getExtension().isRepublishToDlq()))
+					.build();
```
Comment (garyrussell) [inline spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java]: This could be refactored to use `buildRetryTemplateIfRetryEnabled()` instead of the builder.

Comment (garyrussell) [inline spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java]: Something like:

```java
RetryOperationsInterceptor retryInterceptor = RetryInterceptorBuilder.stateless()
	.retryOperations(buildRetryTemplateIfRetryEnabled(properties))
	.recoverer(determineRecoverer(baseQueueName, properties.getExtension().getPrefix(),
			properties.getExtension().isRepublishToDlq()))
	.build();
```

Comment (garyrussell) [inline spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java]: Perhaps (in AB), we could extract the body...

	if (maxAttempts > 1) {
		buildRetryTemplate(properties);
 	}

And then call `buildRetryTemplate(properties)` from here, since we always need a retry template even in maxAttempts is 1 (and we want to republish errors).

**Comment by markfisher** [comment]
LGTM

File: pom.xml [hunk 2]
```diff
@@ -22,6 +22,11 @@
 				<type>pom</type>
 				<scope>import</scope>
 			</dependency>
+			<dependency>
```
Comment (markfisher) [inline pom.xml]: why is this here?

Comment (mbogoevici) [inline pom.xml]: It's establishing the version of `spring-cloud-stream-binder-rabbit` that child modules will use. Typically dependencies are managed in the parent or a `-dependencies` artifact, but since the project is too small, it doesn't warrant one.

Comment (markfisher) [inline pom.xml]: ok

**Comment by markfisher** [comment]
squashed and merged as https://github.com/spring-cloud/spring-cloud-stream-binder-rabbit/commit/e2e02c16939ac960ae3346d3a794a3c75dafd179
